extensions/wikia/PhalanxII/models/PhalanxContentModel.php

```diff
@@ -38,18 +38,17 @@ public function displayBlock() {
 		$this->wg->Out->setArticleRelated( false );
 		$this->wg->Out->addHTML( Html::openElement( 'div', [ 'id' => 'spamprotected_summary' ] ) );
 		$this->wg->Out->addWikiMsg( 'spamprotectiontext' );
-		$this->wg->Out->addHTML( Html::element( 'p', [], wfMessage( 'phalanx-stats-table-id' )->text() . " #{$this->block->id}" ) );
-		$this->wg->Out->addWikiMsg( 'spamprotectionmatch', "<nowiki>{$this->block->text}</nowiki>" );
+		$this->wg->Out->addHTML( Html::element( 'p', [], wfMessage( 'phalanx-stats-table-id' )->text() . " #{$this->block->getId()}" ) );
+		$this->wg->Out->addWikiMsg( 'spamprotectionmatch', "<nowiki>{$this->block->getText()}</nowiki>" );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

[project-id]: wikia:app